tonic-build/src/prost.rs

```diff
@@ -133,14 +133,22 @@ fn is_google_type(ty: &str) -> bool {
     ty.starts_with(".google.protobuf")
 }
 
-struct ServiceGenerator {
+/// A service generator compatible with `prost_build`'s one, to extend its code generator.
+///
+/// This is used internally by Tonic build, but also when the codegen output is not to be handled
+/// by prost or tonic; e.g. such as in a flow for a `protoc` code generation plugin. When compiling
+/// as part of a `build.rs` file, instead use [`compile_protos()`].
+#[derive(Debug)]
+pub struct ServiceGenerator {
     builder: Builder,
     clients: TokenStream,
     servers: TokenStream,
 }
 
 impl ServiceGenerator {
-    fn new(builder: Builder) -> Self {
+    /// Create a new service generator from a configured builder, ready to be passed to
+    /// `prost_build`'s `Config::service_generator`.
+    pub fn new(builder: Builder) -> Self {
```

Comment:
I think maybe we should just add a method to builder to get this type

`Builder::service_generator`

What do you think?

Reply:
@LucioFranco changed to expose only a `service_generator` function. It also directly returns the boxed type since:
- that's what takes prost
- ServiceGenerator does not needs to be pub
- its fields were not pub anyways